pycaret/internal/pycaret_experiment/pycaret_experiment.py

```diff
@@ -447,287 +530,46 @@ def pull(self, pop=False) -> pd.DataFrame:  # added in pycaret==2.2.0
     def X(self):
         """Feature set."""
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            if self.target_param:
-                return self.dataset.drop(self.target_param, axis=1)
-            else:
-                return self.dataset  # For unsupervised: dataset == X
-        else:
-            X = self.dataset.drop(self.target_param, axis=1)
-            if X.empty and self.fe_exogenous is None:
-                return None
-            else:
-                # If X is not empty or empty but self.fe_exogenous is provided
-                # Return X instead of None, since the index can be used to
-                # generate features using self.fe_exogenous
-                return X
+        return self.dataset
 
     @property
-    def y(self):
-        """Target column."""
-        if self.target_param:
-            return self.dataset[self.target_param]
+    def dataset_transformed(self):
+        """Transformed dataset."""
+        return self.train_transformed
 
     @property
     def X_train(self):
         """Feature set of the training set."""
-        if self.target_param is not None:
-            # Supervised Learning
-            X_train = self.train.drop(self.target_param, axis=1)
-        else:
-            # Unsupervised Learning
-            X_train = self.train
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            return X_train
-        else:
-            if X_train.empty and self.fe_exogenous is None:
-                return None
-            else:
-                # If X_train is not empty or empty but self.fe_exogenous is provided
-                # Return X_train instead of None, since the index can be used to
-                # generate features using self.fe_exogenous
-                return X_train
-
-    @property
-    def X_test(self):
-        """Feature set of the test set."""
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            if self.target_param is not None:
-                X_test = self.test.drop(self.target_param, axis=1)
-            else:
-                # Unsupervised Learning
-                X_test = self.test
-        else:
-            # Use index for y_test (idx 2) to get the data
-            test = self.dataset.loc[self.idx[2], :]
-            X_test = test.drop(self.target_param, axis=1)
-
-        if self._ml_usecase != MLUsecase.TIME_SERIES:
-            return X_test
-        else:
-            if X_test.empty and self.fe_exogenous is None:
-                return None
-            else:
-                # If X_test is not empty or empty but self.fe_exogenous is provided
-                # Return X_test instead of None, since the index can be used to
-                # generate features using self.fe_exogenous
-                return X_test
-
-    @property
-    def y_train(self):
-        """Target column of the training set."""
-        if self.target_param:
-            return self.train[self.target_param]
+        return self.train
```

Comment:
Would this return the y values as well?

Reply:
this is overriden in `SupervisedExperiment` to return only X values